lib/base.php

```
@@ -542,9 +542,14 @@ public static function init() {
 		\OC::$server->getEventLogger()->log('autoloader', 'Autoloader', $loaderStart, $loaderEnd);
 		\OC::$server->getEventLogger()->start('boot', 'Initialize');
 
-		// set some stuff
-		//ob_start();
-		error_reporting(E_ALL | E_STRICT);
+		/**
+		 * Set error reporting and debug level
+		 * FIXME: E^DEPRECATED is exluded here due to a behavioural change in
+		 *        PHP 5.6 which will result in valid requests throwing deprecation
+		 *        notices. Remove this hack once PHP 5.6 support is phased out.
+		 * @link https://github.com/owncloud/core/issues/16014
+		 */
+		error_reporting(E_ALL ^ E_DEPRECATED | E_STRICT);
```

> This will break some AJAX requests :-)

Then fix the problem in your dev environment and document it, rather than simply ignore the error message and then have a :boom: 1 year later and have no clue about the cause, because the deprecation warning doesn't show up anymore. Or am I a totally wrong path?